roles/kernel-upgrade/tasks/reboot.yml

```diff
@@ -0,0 +1,30 @@
+---
+
+# Reboot the machine gets more complicated as we want to support bastion hosts. A simple wait_for task would not work
+# as we can not directly reach the hosts (except the bastion). In case a basion is used, we first check for it to come
+# back. After it is back, we check for all the hosts by delegating to the bastion.
+
+- name: Rebooting server
+  shell: nohup bash -c "sleep 5 && shutdown -r now 'Reboot required for updated kernel'" &
```

Comment:
Yeah I know this page and I used this multiple times already. It does not work with bastion hosts in-between however. I don't remember what the problem was however.